SubmersedVR/VR/VRQuickSlots.cs

```diff
@@ -102,6 +107,38 @@ int DetermineSlot(float angle)
                 this.target = quickSlots;
                 this.Init(this.target);
             }
+            if (this.target != null)
+            {
+                //this.HandleInput();
+                int i = 0;
+                int num = this.icons.Length;
+                while (i < num)
+                {
+                    uGUI_ItemIcon uGUI_ItemIcon = this.icons[i];
+                    icons[i].rectTransform.LookAt(Camera.current.transform.position);
+                    backgrounds[i].rectTransform.LookAt(Camera.current.transform.position);
+                    if (!(uGUI_ItemIcon == null))
+                    {
+                        float slotProgress = this.target.GetSlotProgress(i);
+                        float slotCharge = this.target.GetSlotCharge(i);
+                        InventoryItem slotItem = this.target.GetSlotItem(i);
+                        uGUI_ItemIcon.SetBarValue(TooltipFactory.GetBarValue(slotItem));
+                        if (slotProgress < 1f)
+                        {
+                            uGUI_ItemIcon.SetProgress(slotProgress, FillMethod.Radial);
+                        }
+                        else if (slotCharge > 0f)
+                        {
+                            uGUI_ItemIcon.SetProgress(slotCharge, FillMethod.Vertical);
+                        }
+                        else
+                        {
+                            uGUI_ItemIcon.SetProgress(1f, FillMethod.None);
+                        }
+                    }
+                    i++;
+                }
+            }
```

Comment:
I see you copied parts of the decompiled code from uGUI_QuickSlots.Update().
Do you think this could be solved by just calling `base.Update()` here? That would be perfect.

I prefer to find ways to mod without copying too much code from the original game to avoid copyright issues.
Also the decompiled code looks bad. This should be a for loop and this is weird: `if (!(uGUI_ItemIcon == null))`.